### PR TITLE
yarn package manager + eslint & vscode configs

### PR DESCRIPTION
**Pull request by MadSandwich**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[*.{js,json,yml}]
+charset = utf-8
+indent_style = space
+indent_size = 2
```
Comment (VitaliLebedzeu) [inline .editorconfig]:
```suggestion
charset = utf-8
indent_style = space
indent_size = 4
insert_final_newline = true
end_of_line = lf
trim_trailing_whitespace = true

[*.{json,yml}]
indent_size = 2
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (VitaliLebedzeu) [inline .gitignore:8]: perhaps we should split ignores and comment responsible area for each?

File: .yarn/sdks/eslint/bin/eslint.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+if (existsSync(absPnpApiPath)) {
+  if (!process.versions.pnp) {
+    // Setup the environment to be able to require eslint/bin/eslint.js
+    require(absPnpApiPath).setup();
+  }
+}
+
+// Defer to the real eslint/bin/eslint.js your application uses
+module.exports = absRequire(`eslint/bin/eslint.js`);
```
Comment (VitaliLebedzeu) [inline .yarn/sdks/eslint/bin/eslint.js:20]: Could we move it to a common method?

File: .yarn/sdks/eslint/lib/api.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+const absPnpApiPath = resolve(__dirname, relPnpApiPath);
+const absRequire = (createRequire || createRequireFromPath)(absPnpApiPath);
+
+if (existsSync(absPnpApiPath)) {
+  if (!process.versions.pnp) {
+    // Setup the environment to be able to require eslint
+    require(absPnpApiPath).setup();
+  }
+}
```
Comment (VitaliLebedzeu) [inline .yarn/sdks/eslint/lib/api.js:17]: look at the comment above

File: .yarn/sdks/eslint/package.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "eslint",
+  "version": "8.22.0-sdk",
+  "main": "./lib/api.js",
+  "type": "commonjs"
```
Comment (VitaliLebedzeu) [inline .yarn/sdks/eslint/package.json:5]: We would write using commonjs?

File: .gitignore (hunk 2 above)
Comment (MadSandwich) [reply .gitignore:8]: fixed

File: .yarn/sdks/eslint/bin/eslint.js (hunk 3 above)
Comment (MadSandwich) [reply .yarn/sdks/eslint/bin/eslint.js:20]: yarn related

File: .yarn/sdks/eslint/lib/api.js (hunk 4 above)
Comment (MadSandwich) [reply .yarn/sdks/eslint/lib/api.js:17]: yarn related

File: .yarn/sdks/eslint/package.json (hunk 5 above)
Comment (MadSandwich) [reply .yarn/sdks/eslint/package.json:5]: yarn related

**Review by VitaliLebedzeu — APPROVED** [review]

**Pull request merged**